autosklearn/evaluation/train_evaluator.py

```diff
@@ -437,21 +439,25 @@ def fit_predict_and_loss(self, iterative: bool = False) -> None:
                         w / sum(opt_fold_weights) for w in opt_fold_weights
                     ]
 
-                    # train_losses is a list of either scalars or dicts. If it contains
-                    # dicts, then train_loss is computed using the target metric
-                    # (self.metric).
-                    if all(isinstance(elem, dict) for elem in train_losses):
+                    if len(self.metrics) == 1:
                         train_loss = np.average(
                             [
-                                train_losses[i][str(self.metric)]
+                                train_losses[i][str(self.metrics[0])]
                                 for i in range(self.num_cv_folds)
                             ],
                             weights=train_fold_weights_percentage,
                         )
                     else:
-                        train_loss = np.average(
-                            train_losses, weights=train_fold_weights_percentage
-                        )
+                        train_loss = [
```

Comment:
We could also always execute l452-460 and then reduce to a float (in case of a single metric) afterwards.

Reply:
Great idea!